src/backend/ctrl_flow_struct/mod.rs

```diff
@@ -0,0 +1,466 @@
+//! Recovers high-level control-flow constructs from a control-flow graph.
+//! Implements the algorithm described in
+//! [*No More Gotos*](https://doi.org/10.14722/ndss.2015.23185)
```

Comment:
Include a comment for every submodule, what they provide and how it helps this algorithm. Essentially giving a high level picture of how the different parts of this module work together.

Reply:
Does that have to be inline here or can it just be in the documentation for each module?